samples/Infrastructure/Storage.fs

```diff
@@ -345,3 +346,25 @@ module Sql =
             let a = Arguments(p)
             let connection = connect log (a.ConnectionString, a.Schema, a.Credentials, a.AutoCreate) |> Async.RunSynchronously
             StorageConfig.Sql(SqlStreamStoreContext(connection, batchSize = a.BatchSize), cacheStrategy cache, unfolds)
+     module MsgDb =
+             open Equinox.MessageDb
+             type [<NoEquality; NoComparison>] Parameters =
+                 | [<AltCommandLine "-c"; Mandatory>] ConnectionString of string
+                 | [<AltCommandLine "-A">]       AutoCreate
+                 | [<AltCommandLine "-b">]       BatchSize of int
+                 interface IArgParserTemplate with
+                     member a.Usage = a |> function
+                         | ConnectionString _ -> "Database connection string"
+                         | AutoCreate _ ->       "AutoCreate schema"
+                         | BatchSize _ ->        "Maximum number of Events to request per batch. Default 500."
+             type Arguments(p : ParseResults<Parameters>) =
+                 member _.ConnectionString =     p.GetResult ConnectionString
+                 member _.BatchSize =            p.GetResult(BatchSize, 500)
+             let connect (log : ILogger) (connectionString: string) =
+                 log.Information("MessageDB Connection {connectionString}", connectionString)
+                 MessageDbConnector(connectionString).Establish()
+             let config (log : ILogger) cache (p : ParseResults<Parameters>) =
+                 let a = Arguments(p)
+                 let connection = connect log a.ConnectionString |> Async.AwaitTask |> Async.RunSynchronously
```

Comment:
in general, AwaitTaskCorrect is used consistently so no need to second guess semantics

Reply:
[Still waiting](https://github.com/fsharp/fslang-suggestions/issues/840)
![Still waiting](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fwww.memecreator.org%2Fstatic%2Fimages%2Fmemes%2F5025828.jpg&f=1&nofb=1&ipt=0a1bee1852aec01306f30542093733885f5780999756c7d4ce467f2a585a8912&ipo=images)